server/unittools.cpp

```diff
@@ -1508,6 +1537,16 @@ void transform_unit(struct unit *punit, const struct unit_type *to_unit,
     punit->moves_left = punit->moves_left * unit_move_rate(punit) / old_mr;
   }
 
+  /* Update unit fuel state */
+  if (utype_fuel(to_unit) && utype_fuel(old_type)) {
+    /* Keep the amount of fuel used constant, except don't kill the unit */
+    int delta = utype_fuel(to_unit) - utype_fuel(old_type);
+    punit->fuel = MAX(punit->fuel + delta, 1);
```

Comment:
This looks strange. If I have 1/10 fuel, I end up with 91/100. If I have 100/100 fuel, I end up with 1/10.

Reply:
No you don't?  In your second case, `delta = 10 - 100` = -90, so you end up with `MAX(100 - 90, 1)` i.e. 10/10.  (A fully-fueled unit should always remain fully-fueled.)

Reply:
Eeeh right.